app/Http/Controllers/Setup/Controller.php

```diff
@@ -164,8 +164,18 @@ public function save_config() {
         }
 
         if(self::is_heroku()) {
-            self::write_config_value($config, 'QUEUE_CONNECTION', 'database');
-
+            if (!empty(env('REDIS_URL'))) {
+                self::write_config_value($config, 'QUEUE_CONNECTION', 'redis');
+                if (!empty(env('REDIS_SESSION'))) {
```

Comment:
What is `REDIS_SESSION`? Is that a different URL that the redis URL? When would it be set vs not set? Some inline comments here would be helpful.

Reply:
it's just a chosen environment variable, who's mere presence in conjunction with `REDIS_URL` (see above) causes the session config to be set to 'redis'

Reply:
Same for the `REDIS_CACHE`. I think I should maybe put these into README.md then

It'd be needed to be run prior to setup completion

Reply:
I'm not sure I understand the use case for this, can you explain how someone would use it and why?

Reply:
For a queue it makes sense to use REDIS rather than a database. It has [`rpoplpush`](https://redis.io/commands/rpoplpush) which can ensure that in-progress items are retrievable if a worker dies.

For a session, if you have many servers and want to put private data there; then you might want Redis. It's not on by default.

For a cache accessing single values through Redis might be beneficial for some workloads. It's also not on by default.

Reply:
But what's the value of setting this extra config option vs setting SESSION_DRIVER directly?

Reply:
if someone wants to deliberately set it, they can. All this does is at setup-time indicate "Hey, I'd like these optional things on". If someone does not indicate this, then I think it would be silly to not have cookie sessions and filesystem cache which is wiped out on reboot. These are things that I think are valuable, but not for the majority of people, which is why only the `QUEUE_CONNECTION` is deliberately turned on when a `REDIS_URL` is specified

Reply:
It's also why they are in a separate place. I'm happy to drop the commit. But I won't leave them unguarded